vllm/distributed/kv_transfer/kv_connector/factory.py

```diff
@@ -66,6 +66,22 @@ def create_connector(
         # We build separately to enforce strict separation
         return connector_cls(config, role)
 
+    @classmethod
+    def get_connector_class_by_name(
+        cls, connector_name: str
+    ) -> type[KVConnectorBaseType] | None:
+        """Get a registered connector class by name.
+
+        Args:
+            connector_name: Name of the registered connector.
+
+        Returns:
+            The connector class if found, None otherwise.
+        """
+        if connector_name not in cls._registry:
+            return None
```

Comment:
as above, I think we can raise like in `get_connector_class` if the connector isn't registered, as registration should happen once at startup